File: 백준/Gold/11404. 플로이드/플로이드.java
import java.io.*;
import java.util.Arrays;


public class Main{
    public static void main(String[] args) throws IOException{

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int n = Integer.parseInt(br.readLine());
        int m = Integer.parseInt(br.readLine());

        int[][] dist = new int[n][n];
        for (int i=0; i<n; i++){
            Arrays.fill(dist[i], 1000000000);
            dist[i][i] = 0;
        }
        for (int i=0; i<m ;i++) {
            String[] input = br.readLine().split(" ");
            int a = Integer.parseInt(input[0]);
            int b = Integer.parseInt(input[1]);
            int c = Integer.parseInt(input[2]);

            if (dist[a-1][b-1] > c) {
                dist[a-1][b-1] = c;
            }
        }

        for(int k=0; k<n; k++) {
            for (int a=0; a<n; a++) {
                for(int b=0; b<n; b++) {
                    dist[a][b] = Math.min(dist[a][b], dist[a][k] + dist[k][b]);
                }
            }
        }

        for (int a=0; a<n; a++) {
            for (int b=0; b<n; b++) {
                if(dist[a][b] == 1000000000)
                    bw.write(0+" ");
                else
                    bw.write(dist[a][b] + " ");
            }
            bw.newLine();;
        }

        bw.flush();
    }
}

File: 백준/Gold/11404. 플로이드/플로이드.py
INF = int(1e9)

if __name__ == '__main__':

    n = int(input())
    m = int(input())

    dist = [[INF] * n for _ in range(n)]

    for idx in range(n):
        dist[idx][idx] = 0

    for _ in range(m):
        str = input().split(" ")
        start = int(str[0])
        adj = int(str[1])
        d = int(str[2])
        if d < dist[start-1][adj-1]:
            dist[start-1][adj-1] = d

    for k in range(n):
        for a in range(n):
            for b in range(n):
                dist[a][b] = min(dist[a][b], dist[a][k] + dist[k][b])

    for start in range(n):
        for adj in range(n):
            if dist[start][adj] == INF:
                print(0, end=" ")
            else:
                print(dist[start][adj], end=" ")

        print()